syntax = "proto3";

package chat;

import "google/protobuf/timestamp.proto";

// Define the ChatRoomMessage message
message ChatRoomMessage {
    int32 id = 1;                             // Primary key
    string content = 2;                       // Message content
    int32 sender_id = 3;                      // Foreign key to sender (User)
    int32 chat_room_id = 4;                   // Foreign key to ChatRoom
    google.protobuf.Timestamp timestamp = 5; // Timestamp of the message (from DB)
}

// Service for ChatRoomMessage-specific operations
service ChatRoomMessageService {
    // Get messages by chat room ID
    rpc GetMessagesByChatRoomId(GetMessagesByChatRoomIdRequest) returns (ListChatRoomMessagesResponse);

    // Create a new chat room message
    rpc CreateMessage(CreateChatRoomMessageRequest) returns (ChatRoomMessage);
}

// Request and Response messages for ChatRoomMessage
message GetMessagesByChatRoomIdRequest {
    int32 chat_room_id = 1; // Chat room ID
}

message CreateChatRoomMessageRequest {
    string content = 1;      // Message content
    int32 sender_id = 2;     // Foreign key to sender (User)
    int32 chat_room_id = 3;  // Foreign key to ChatRoom
}

message ListChatRoomMessagesResponse {
    repeated ChatRoomMessage messages = 1; // List of chat room messages
}
